### ISSUE TITLE
Redux + TabNavigator Not Updating Unless Changing Tab

### ISSUE THREAD

**Issue by zealios** [react-community/react-navigation#1229]
Hi,

this.props.screenProps.count only get updated when I'm changing tab. It work just fine if I change my code from TabNavigator to StackNavigator.

Sample code below:

```javascript
import React from 'react';

import {
  AppRegistry,
  View,
  Text,
  Button
} from 'react-native';

import {
  combineReducers,
  createStore
} from 'redux';

import {
  connect,
  Provider
} from 'react-redux';

import {
  TabNavigator,
  addNavigationHelpers
} from 'react-navigation';

class Test1 extends React.Component {
  render() {
    return (
      <View>
		<Text>Test 1, count: {this.props.screenProps.count}</Text>
		<Button title="increase" onPress={() => {this.props.screenProps.increment()}} />
	  </View>
    );
  }
}

class Test2 extends React.Component {
  render() {
    return (
      <View>
		<Text>Test 2, count: {this.props.screenProps.count}</Text>
		<Button title="decrement" onPress={() => {this.props.screenProps.decrement()}} />
	  </View>
    );
  }
}

class Test3 extends React.Component {
  render() {
    return (
      <View>
		<Text>Test 3, count: {this.props.screenProps.count}</Text>
		<Button title="reset" onPress={() => {this.props.screenProps.reset()}} />
	  </View>
    );
  }
}

const AppNavigator = TabNavigator({
  Test1: {
    screen: Test1
  },
  Test2: {
    screen: Test2
  },
  Test3: {
    screen: Test3
  }
});

const navReducer = (state, action) => {
  const nextState = AppNavigator.router.getStateForAction(action, state);
  return nextState || state;
};

const countReducer = (state = 0, action) => {
  switch (action.type) {
    case 'INCREMENT':
      return state + 1;
    case 'DECREMENT':
      return state - 1;
    case 'RESET':
      return 0;
    default:
      return state;
  }
};

const appReducer = combineReducers({
  nav: navReducer,
  count: countReducer
});

class App extends React.Component {
  render() {
    const newProps = { ...this.props };
    delete newProps.dispatch;
    delete newProps.nav;

    return (
      <AppNavigator 
        navigation={addNavigationHelpers({
          dispatch: this.props.dispatch,
          state: this.props.nav
      	})} 
      	screenProps={{...newProps}}
  	  />
    );
  }
}

const mapStateToProps = (state) => ({
  nav: state.nav,
  count: state.count
});

const mapDispatchToProps = (dispatch) => {
  let actionCreators = {
    increment: () => {
      dispatch({
        type: 'INCREMENT'
      })
    },
    decrement: () => {
      dispatch({
        type: 'DECREMENT'
      })
    },
    reset: () => {
      dispatch({
        type: 'RESET'
      })
    },
  }

  return {
    ...actionCreators,
    dispatch
  };
};

const AppWithNavigationState = connect(mapStateToProps, mapDispatchToProps)(App);

class Root extends React.Component {
  store = createStore(appReducer);

  render() {
    return (
      <Provider store={this.store}>
        <AppWithNavigationState />
      </Provider>
    );
  }
}

AppRegistry.registerComponent('Kalam', () => Root);
```

**Comment by grabbou** [comment]
CC: @satya164

**Comment by callagga** [comment]
thanks @zealios for posting, as whilst it may highlight an issue, I've managed to use your code here to get my first working react-native + react-navigation + redux sample app to run!

**Comment by zealios** [comment]
FYI @callagga 

As workaround for now I'm using component state to give instance update to UI (same time I still do store dispatch) and sync it later with store state once event componentWillReceiveProps being triggered.

**Comment by zealios** [comment]
@grabbou @satya164 

Tested and this issue solved in 1.0.0-beta.11 release.

**Comment by troylutton** [comment]
I totally agree with @callagga. This was a great small yet complete example of how to use redux with react navigation. The example in the react navigation repo doesn't contain actions
